File: src/agents/Flutter/nexus.py
from src.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class NexusPersona(BaseActivePersona):
    """
    Core: PhD in Distributed Systems & Transport Layers (Flutter) 🌐
    Especialista em resiliência de rede, protocolos HTTP/gRPC e timeouts Dart.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Nexus", "🌐", "PhD Network Engineer", "Flutter"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Camadas de Transporte Flutter...")
        
        audit_rules = [
            {'regex': r"http\.get\(", 'issue': 'Resiliência: Uso da biblioteca http padrão. Considere Dio para melhor tratamento de erros e interceptores.', 'severity': 'medium'},
            {'regex': r"connectTimeout\s*:\s*null", 'issue': 'Fragilidade: Timeout de conexão infinito detectado.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.dart',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "connectTimeout: null" in content:
            return f"Risco de Bloqueio: O objetivo '{objective}' exige reatividade. Em '{file}', a falta de timeout pode paralisar a 'Orquestração de Inteligência Artificial' em redes instáveis."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em redes e transporte Flutter."

File: src_local/agents/Bun/System/nexus.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * 🌐 Dr. Nexus — PhD in Bun Networking & Bun.serve Resilience
 * Especialista em Bun.serve, WebSockets nativos e resiliência HTTP Bun.
 */
export class NexusPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Nexus";
        this.emoji = "🌐";
        this.role = "PhD Bun Network Resilience Engineer";
        this.phd_identity = "Networking & Bun.serve Resilience Architecture";
        this.stack = "Bun";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.tsx', '.js'],
            rules: [
                { regex: /fetch\s*\([^)]*\)(?![\s\S]{0,100}signal|AbortController)/, issue: 'Fragilidade: fetch() sem timeout/AbortController no Bun PhD.', severity: 'high' },
                { regex: /Bun\.serve\([^)]*websocket[^)]*\)(?![\s\S]{0,200}close|error)/, issue: 'WebSocket Frágil: Bun.serve WebSocket sem handler close/error PhD.', severity: 'high' },
                { regex: /new\s+WebSocket\([^)]*\)(?![\s\S]{0,200}onerror)/, issue: 'Conexão Frágil: WebSocket sem handler de erro no Bun PhD.', severity: 'high' },
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && /fetch\s*\(/.test(content) && !/AbortController|signal/.test(content)) {
            return {
                file, severity: "HIGH",
                issue: `Fragilidade: O objetivo '${objective}' exige resiliência. Em '${file}', fetch sem timeout ameaça o Bun.serve PhD.`,
                context: "fetch without timeout detected"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Nexus (Bun): Analisando conectividade e resiliência de rede para ${objective}.`,
            context: this.name
        };
    }

    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em redes e resiliência HTTP Bun.`;
    }
}
